### PR TITLE
WIP: Updating Factory To Be More Generic (Not Framework Specific).

### PR DESCRIPTION
**Pull request by Fenikkusu**
This PR:

- [x] Adds new feature ...
- [ ] Covered by tests
- [ ] Doc PR: https://github.com/infection/site/pull/XXX

Preliminary PR For Factory Clean Up Per Disccusions In PR #457. This PR is to clean up the TestFramework factory to not be framework specific. This will allow new frameworks to be added without the need to update the factory with the new framework.

### CODE REVIEW

**Comment by Fenikkusu** [comment]
Does it make sense to standardize class names as part of this. I see no reason to repeat the framework name in the class name when the namespace is already doing that. For example:

```
\Infection\TestFramework\PhpUnit\ExtraOptions
\Infection\TestFramework\PhpSpec\ExtraOptions
\Infection\TestFramework\Codeception\ExtraOptions
```

instead of

```
\Infection\TestFramework\PhpUnit\PhpUnitExtraOptions
\Infection\TestFramework\PhpSpec\PhpSpecExtraOptions
\Infection\TestFramework\Codeception\CodeceptionExtraOptions
```

**Comment by Fenikkusu** [comment]
Looking to get some feedback on this most recent run. I've gotten the factory and name spacing cleaned up. One thing I noticed is that the factory is being initialized with params specifically for PhpUnit. This doesn't make sense to me, so I started to strip those out, with the intention of initialization in the actual relative classes. This begs the question of what makes the most sense to access them. They were originally in the container, so I can pull them from there as well, but I'd either have to pass it through or create a new instance, neither of which seems appropriate. Thoughts?

**Comment by Fenikkusu** [comment]
Looking to get some feedback on this most recent run. I've gotten the factory and name spacing cleaned up. One thing I noticed is that the factory is being initialized with params specifically for PhpUnit. This doesn't make sense to me, so I started to strip those out, with the intention of initialization in the actual relative classes. This begs the question of what makes the most sense to access them. They were originally in the container, so I can pull them from there as well, but I'd either have to pass it through or create a new instance, neither of which seems appropriate. Thoughts?

**Comment by borNfreee** [comment]
Could you please give more details, probably with code examples? I'm not sure I understand the problem you are trying to solve.

> This doesn't make sense to me, so I started to strip those out, with the intention of initialization in the actual relative classes.

What do you mean by `actual relative classes`?

As for me, it's ok that different test framework adapters require different dependencies. Keeping this in mind I don't know how to make the factory class framework independent. We can leave dependency management for Container itself, but in this case, we will still need to create additional container service when new test framework is implemented.

If this is going to be so complex, let's answer to the question: does it worth refactoring at all? We are adding test frameworks not so often ;)

**Comment by Fenikkusu** [comment]
Well, keep in mind that a portion of the refactor comes more from the FQNS changes. If I were to just move all the creations directly to the container, that would eliminate a chunk of the complexity of the change.

By `actual relative classes`, I mean, primarily, the PhpUnit Initial Builder. It has dependencies on the xml configuration helper. Rather than passing this through the to the Factory, I think it makes more sense to let the Builder just directly pull from the container. If that makes sense.

File: src/TestFramework/Factory.php [hunk 1]
```diff
@@ -67,60 +48,73 @@ public function __construct(
...
+        $configPath = $this->configLocator->locate($adapterName);
 
-            return new PhpSpecAdapter(
-                new TestFrameworkFinder(TestFrameworkTypes::PHPSPEC),
-                new PhpSpecInitialConfigBuilder($this->tmpDir, $phpSpecConfigPath, $skipCoverage),
-                new PhpSpecMutationConfigBuilder($this->tmpDir, $phpSpecConfigPath, $this->projectDir),
-                new PhpSpecArgumentsAndOptionsBuilder(),
-                $this->versionParser
-            );
+        $baseNamespace = __NAMESPACE__ . '\\' . $adapterName . '\\';
```
Comment (BackEndTea) [inline src/TestFramework/Factory.php]: I prefer the old if, else statements, it does add mroe boilerplate code, but in all reality there won't be much test frameworks being added. This is too 'magical', and makes is harder to reason about what is going on.

**Comment by Fenikkusu** [comment]
Okie Dokie

**Pull request closed without merging**